### ISSUE TITLE
Coloring of strings could be misleading

### ISSUE THREAD

**Issue by nicolaiskogheim** [atom/language-yaml#17]
I just had an issue with a line of code like this:
```
script: "coverage report --omit="*__init__.py""
```

It took me a little while to figure out my copypaste error with the nested double quotes.

This is the same line with githubs .yml syntax coloring
```yml
script: "coverage report --omit="*__init__.py""
```

There it is very easy to see where I went wrong.

Screen from atom
![screenshot](https://www.dropbox.com/s/txfjjt1dkm3g4k7/Screenshot%202014-10-22%2005.03.07.png?dl=1)

**Comment by floscher** [comment]
> This is the same line with githubs .yml syntax coloring
```yaml
script: "coverage report --omit="*__init__.py""
```
There it is very easy to see where I went wrong.

GitHub also [uses this grammar for syntax highlighting](https://github.com/github/linguist/blob/master/.gitmodules#L88), so by now it has the same appearance as in atom ;-)

Currently e.g. JSON-highlighting still shows the behaviour you probably meant:
![](https://cloud.githubusercontent.com/assets/3904348/5847866/78ce4898-a1d2-11e4-9d84-3289ce398bcb.png)

**Comment by nicolaiskogheim** [comment]
Good catch! You are absolutely right. Pity that Github has this issue now.

**Comment by 50Wliu** [comment]
I took a look at this and I don't believe this is a bug.  The quoted parts get `string.quoted.double.yaml`, while the unquoted part (`*__init__.py`) gets `string.unquoted.yaml`.  Those two can be styled differently.

I'm closing this, but please feel free to correct me if I'm wrong.

**Comment by nicolaiskogheim** [comment]
Ah, okay. Thank you for taking a look :)

**Comment by 50Wliu** [comment]
After some thought I've decided to reopen this as we really shouldn't be highlighting that inner segment as unquoted strings.
